document/core/exec/runtime.rst

```diff
@@ -678,51 +697,56 @@ This definition allows to index active labels surrounding a :ref:`branch <syntax
 
 
-.. index:: ! throw context, exception, throw address, catch block
+.. index:: ! throw context, tag, throw address, catch block, handler, exception
 .. _syntax-ctxt-throw:
 
 Throw Contexts
 ..............
 
 In order to specify the reduction of |TRY| blocks
-with the help of the administrative instructions |THROWADDR|, |REFEXNADDR|, and |CATCHN|,
+with the help of the administrative instructions |THROWadm|, |CATCHadm|, |DELEGATEadm|, and |CAUGHTadm|,
 the following syntax of *throw contexts* is defined, as well as associated structural rules:
 
 .. math::
    \begin{array}{llll}
    \production{(throw contexts)} & \XT &::=&
      \val^\ast~[\_]~\instr^\ast \\ &&|&
      \LABEL_n\{\instr^\ast\}~\XT~\END \\ &&|&
-     \CATCHN_n\{\instr^\ast\}~\XT~\END \\ &&|&
+     \CAUGHTadm\{a~\val^\ast\}~\XT~\END \\ &&|&
      \FRAME_n\{F\}~\XT~\END \\
    \end{array}
 
-Throw contexts allow matching the program context around a throw instruction up to the nearest enclosing :math:`\CATCHN` handler, thereby selecting the exception handler responsible for an exception.
+Throw contexts allow matching the program context around a throw instruction up to the innermost enclosing |CATCHadm| or |DELEGATEadm|, thereby selecting the exception |handler| responsible for an exception, if one exists.
+
+.. note::
+   Contrary to block contexts, throw contexts don't skip over handlers.
+
+   Since handlers are not included above, there is always a unique maximal throw context to match the reduction rules.
+   This basically means that :math:`\CAUGHTadm {..} instr^\ast \END` is not a potential catching block for exceptions thrown by :math:`\instr^\ast`, since it is inside a |handler|'s block.
 
 .. note::
-   For example, catching a simple :ref:`throw <exec-throw>` in a :ref:`try block <exec-try>` would be as follows.
+   For example, catching a simple :ref:`throw <exec-throw>` in a :ref:`try block <exec-try-catch>` would be as follows.
 
    .. math::
       \begin{array}{ll}
-      & \hspace{-5ex} S;~F;~\val^n~(\TRY~\X{bt}~\THROW~x~\CATCH~\RETURN~\END) \\
-      \stepto & S;~F;~\CATCHN_1\{\RETURN\}~(\LABEL_1 \{\}~\val^n~\THROWADDR~a~\END)~\END \\
+      & \hspace{-5ex} S;~F;~\val^n~(\TRY~\X{bt}~\THROW~x~\CATCH~x~\RETURN~\END) \\
+      \stepto & S;~F;~\LABEL_n\{\} (\CATCHadm\{a~\RETURN\}~~\val^n~\THROWadm~a~\END)~\END \\
       \end{array}
 
-   :ref:`Handling <exec-throwaddr>` the thrown exception address :math:`a` in the throw context
-   :math:`T=\LABEL_1 \{\}[\_]~\END`, with the exception handler :math:`\CATCHN_1\{\RETURN\}` gives:
+   :ref:`Handling <exec-throwadm>` the thrown exception with tag address :math:`a` in the throw context
+   :math:`T=[\_]`, with the exception handler :math:`H=\CATCHadm\{a~\RETURN\}` gives:
```

Comment:
Where is `H` used/coming from?

Reply:
In line 572 we had specified (already in the previous version of the spec):

> * The meta variable :math:`H` ranges over exception handlers where clear from context.

Reply:
Interesting suggestion. But probably something we should discuss separately.

Reply:
> should these metavariables be listed somewhere, like an index of metavariables?

Interesting suggestion. But probably something we should discuss separately.